def fact(n):
    '''
    Calculate 1*2*...*n

    >>> fact(1)
    1
    >>> fact(10)
    3628800
    >>> fact(-1)
    Traceback (most recent call last):
      File "doctest-test.py", line 11, in <module>
        print(fact(-1))
      File "doctest-test.py", line 6, in fact
        raise ValueError()
    ValueError
    '''
    if n < 1:
        raise ValueError()
    if n == 1:
        return 1
    return n * fact(n - 1)

    if __name__ == '__main__':
    import doctest
    doctest.testmod()
